plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java

```diff
@@ -122,27 +121,26 @@ private void addDependenciesToProject(IProject project, IProgressMonitor monitor
     SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
 
     // Create a lib folder
-    IFolder libFolder = project.getFolder("lib");
+    IFolder libFolder = project.getFolder("lib"); //$NON-NLS-1$
     if (!libFolder.exists()) {
       libFolder.create(true, true, subMonitor.newChild(10));
     }
 
     // Download the dependencies from maven
     int ticks = 50 / PROJECT_DEPENDENCIES.size();
-    for (Map.Entry<String, String> dependency : PROJECT_DEPENDENCIES.entrySet()) {
-      LibraryFile libraryFile = new LibraryFile(new MavenCoordinates(dependency.getKey(),
-          dependency.getValue()));
+    for (MavenCoordinates dependency : PROJECT_DEPENDENCIES) {
+      LibraryFile libraryFile = new LibraryFile(dependency);
       File artifactFile = null;
       try {
         Artifact artifact = repositoryService.resolveArtifact(libraryFile, subMonitor.newChild(ticks));
         artifactFile = artifact.getFile();
         IFile destFile = libFolder.getFile(artifactFile.getName());
         destFile.create(new FileInputStream(artifactFile), true, subMonitor.newChild(30));
       } catch (CoreException ex) {
-        logger.log(Level.WARNING, "Error downloading " +
-            libraryFile.getMavenCoordinates().toString() + " from maven", ex);
+        logger.log(Level.WARNING, "Error downloading " + //$NON-NLS-1$
+            libraryFile.getMavenCoordinates().toString() + " from maven", ex); //$NON-NLS-1$
       } catch (FileNotFoundException ex) {
-        logger.log(Level.WARNING, "Error copying over " + artifactFile.toString() + " to " +
+        logger.log(Level.WARNING, "Error copying over " + artifactFile.toString() + " to " + //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
The exception is from `new FileInputStream(artifactFile)`, so at that point `artifactFile` is not null. Also, `artifact.getFile()` never returns `null`, so we are good.